src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public Figure getRandomFigure() {
+        Random random = new Random();
+        ColorSupplier cs = new ColorSupplier();
```

Comment:
same, let's make Random and ColorSupplier as class level variables, not to create new instances every time we call getRandomFigure() method